ci/check_skip_system_icu_linking.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
```

Comment:
make this a sh_test in test/dependencies directory, just like other sh_test we does for envoy binary.

Reply:
I have moved it.